internal/provider/resource_cloudflare_pages_project.go

```diff
@@ -170,7 +175,7 @@ func resourceCloudflarePagesProjectRead(ctx context.Context, d *schema.ResourceD
 		return diag.FromErr(fmt.Errorf("error reading cloudflare pages project %q: %w", projectName, err))
 	}
 
-	d.SetId(res.ID)
+	d.SetId(res.Name)
 	d.Set("subdomain", res.SubDomain)
 	d.Set("created_on", res.CreatedOn.Format(time.RFC3339))
 	d.Set("domains", res.Domains)
```

Comment:
to make sure all the state is imported, you'll need to add the `d.Set()` calls for the other various bits of config (production branch, source, deployments, etc)